subsys/bluetooth/controller/ll_sw/ull_iso.c

```diff
@@ -496,13 +519,206 @@ int ull_iso_reset(void)
 	return 0;
 }
 
+#if defined(CONFIG_BT_CTLR_SYNC_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+void *ull_iso_pdu_rx_alloc_peek(uint8_t count)
+{
+	if (count > MFIFO_AVAIL_COUNT_GET(iso_rx)) {
+		return NULL;
+	}
+
+	return MFIFO_DEQUEUE_PEEK(iso_rx);
+}
+
+void *ull_iso_pdu_rx_alloc(void)
+{
+	return MFIFO_DEQUEUE(iso_rx);
+}
+
+void ull_iso_rx_put(memq_link_t *link, void *rx)
+{
+	/* Enqueue the Rx object */
+	memq_enqueue(link, rx, &memq_ull_iso_rx.tail);
+}
+
+void ull_iso_rx_sched(void)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, iso_rx_demux};
+
+	/* Kick the ULL (using the mayfly, tailchain it) */
+	mayfly_enqueue(TICKER_USER_ID_LLL, TICKER_USER_ID_ULL_HIGH, 1, &mfy);
+}
+
+static void iso_rx_demux(void *param)
+{
+	struct ll_conn_iso_stream *cis;
+	struct ll_iso_datapath *dp;
+	struct node_rx_pdu *rx_pdu;
+	isoal_sink_handle_t sink;
+	struct node_rx_hdr *rx;
+	memq_link_t *link;
+
+	do {
+		link = memq_peek(memq_ull_iso_rx.head, memq_ull_iso_rx.tail,
+				 (void **)&rx);
+		if (link) {
+			/* Demux Rx objects */
+			switch (rx->type) {
+			case NODE_RX_TYPE_RELEASE:
+				memq_dequeue(memq_ull_iso_rx.tail, &memq_ull_iso_rx.head, NULL);
+				ll_iso_rx_put(link, rx);
+				ll_rx_sched();
+				break;
+
+			case NODE_RX_TYPE_ISO_PDU:
+				/* Remove from receive-queue; ULL has received this now */
+				(void)memq_dequeue(memq_ull_iso_rx.tail, &memq_ull_iso_rx.head,
+						   NULL);
+
+				rx_pdu = (struct node_rx_pdu *)rx;
+				cis = ll_conn_iso_stream_get(rx_pdu->hdr.handle);
+				dp = cis->datapath_out;
+				sink = dp->sink_hdl;
+
+				if (dp->path_id != BT_HCI_DATAPATH_ID_HCI) {
+					/* If vendor specific datapath pass to ISO AL here,
+					 * in case of HCI destination it will be passed in
+					 * HCI context.
+					 */
+					struct isoal_pdu_rx pckt_meta = {
+						.meta = &rx_pdu->hdr.rx_iso_meta,
+						.pdu  = (union isoal_pdu *) &rx_pdu->pdu[0]
```

Comment:
```suggestion
						.pdu  = (union isoal_pdu *)&rx_pdu->pdu[0]
```

Reply:
> Another issue is that that it is tuned for a Connected ISO build, and does not consider the iso sync receiver.
Yes, exactly. This is why I suggested to add the explaining text, to let it be up to the user to do the math for the applicable use case. I think it would be too complex to calculate with all variations taken into consideration. Also, there might be other deciding factors such as available memory and architecture.